src/interaction/navigationhandler.cpp

```diff
@@ -105,45 +131,29 @@ ghoul::Dictionary NavigationHandler::NavigationState::dictionary() const {
 }
 
 NavigationHandler::NavigationState::NavigationState(const ghoul::Dictionary& dictionary) {
-    const bool hasAnchor = dictionary.hasValue<std::string>(KeyAnchor);
-    const bool hasPosition = dictionary.hasValue<glm::dvec3>(KeyPosition);
-    if (!hasAnchor || !hasPosition) {
-        throw ghoul::RuntimeError(
-            "Position and Anchor need to be defined for navigation dictionary."
-        );
-    }
+    const Parameters p = codegen::bake<Parameters>(dictionary);
 
-    anchor = dictionary.value<std::string>(KeyAnchor);
-    position = dictionary.value<glm::dvec3>(KeyPosition);
+    anchor = p.anchor;
+    position = p.position;
 
-    if (dictionary.hasValue<std::string>(KeyReferenceFrame)) {
-        referenceFrame = dictionary.value<std::string>(KeyReferenceFrame);
-    }
-    else {
-        referenceFrame = anchor;
-    }
-    if (dictionary.hasValue<std::string>(KeyAim)) {
-        aim = dictionary.value<std::string>(KeyAim);
-    }
+    referenceFrame = p.referenceFrame.value_or(anchor);
+    aim = p.aim.value_or(aim);
 
-    if (dictionary.hasValue<glm::dvec3>(KeyUp)) {
-        up = dictionary.value<glm::dvec3>(KeyUp);
+    // @TODO (abock, 2021-03-24) This seems a bit odd;  is it correct that we only want to
```

Comment:
Not completely sure, but I think the idea here is that the yaw and pitch angles does not really make sense unless you also know the up vector. Without an up vector we only know about the forward direction of the camera, and simply don't have sufficient information to compute the camera orientation from the yaw/pith. 

That being said, maybe the up-vector should actually just have a default value

Reply:
Didn't think about that.. i think it makes sense